### PR TITLE
[UNDERTOW-1969] Ignore deprecated certificate retrieval method failure for deprecated parts of Undertow

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/UNDERTOW-1969

### CODE REVIEW

**Review by fl4via — APPROVED** [review]

File: core/src/main/java/io/undertow/server/BasicSSLSessionInfo.java [hunk 1]
```diff
@@ -67,7 +67,15 @@ public BasicSSLSessionInfo(byte[] sessionId, String cypherSuite, String certific
...
             for(java.security.cert.Certificate cert : certCol) {
                 this.peerCertificate[i] = cert;
-                this.certificate[i++] = X509Certificate.getInstance(cert.getEncoded());
+                try {
+                    this.certificate[i] = X509Certificate.getInstance(cert.getEncoded());
+                } catch (CertificateException ce) {
+                    // [UNDERTOW-1969] We don't care about deprecated JDK methods failure caused by the fact newer JDKs
+                    // doesn't support them anymore. "this.certificate" is used only by deprecated method
+                    // {@link SSLSessionInfo.getPeerCertificateChain()} which call should be avoided by API users.
+                    this.certificate[i] = null;
```
Comment (bstansberry) [inline core/src/main/java/io/undertow/server/BasicSSLSessionInfo.java:76]: @ropalka @fl4via This is not a big deal but if this happens shouldn't the result be that this.certificate (the whole array) gets set to null? Then getPeerCertificateChain() will throw SSLPeerUnverifiedException which seems cleaner than returning an array with some null elements.

Comment (bstansberry) [reply core/src/main/java/io/undertow/server/BasicSSLSessionInfo.java:76]: In case you agree I sent up https://github.com/undertow-io/undertow/pull/1255

Comment (ropalka) [reply core/src/main/java/io/undertow/server/BasicSSLSessionInfo.java:76]: I agree.

Comment (fl4via) [reply core/src/main/java/io/undertow/server/BasicSSLSessionInfo.java:76]: @bstansberry  thanks for fixing this!

**Pull request merged**